core/src/main/scala/org/http4s/package.scala

```diff
@@ -136,4 +77,13 @@ package object http4s { // scalastyle:ignore
       case Left(l) => f(l)
     }
   }
+
+  implicit def http4sHttpServiceSemigroup[F[_]: Monad]: Semigroup[HttpService[F]] =
+    new Semigroup[HttpService[F]] {
+      override def combine(x: HttpService[F], y: HttpService[F]) =
+        Kleisli(request => x(request).orElse(y(request)))
+    }
+
+  def FToOptionT[F[_]: Functor]: FunctionK[F, OptionT[F, ?]] =
```

Comment:
Not sure where this should live. It's come in handy a couple of times.

Reply:
This is a good question for the Cats room.  If we can't get a satisfactory cats answer, then we look to `org.http4s.util`, I think.